design-proposals/dgpu-support.md

```diff
@@ -0,0 +1,204 @@
+     * **For EMT** : EMT should be build with 6.11* linux kernel.
+     * **For Ubuntu** : Ubuntu Server 24.04.2, equipped with the 6.11 kernel, has been validated with BMG GPU drivers. DEB packages are available as rolling updates in the official documentation.
+
+   * **NVIDIA**
+     * A new extension will be created to configure and install the NVIDIA GPU Operator using its Helm chart. The NVIDIA GPU Operator automates the management of all NVIDIA software components required to provision GPUs in Kubernetes, including drivers, the Kubernetes device plugin, the NVIDIA Container Runtime, and monitoring tools.
+        ![gpu-operator-extension](images/nvidia-gpu-operator-extension-package.png)
+     * **For EMT** : pre-installed drivers in the immutable OS will be utilized by the GPU Operator.
+     * **For Ubuntu** : the extension package will configure the GPU Operator to detect the OS and install optimal drivers automatically.
+
+1. **SRIOV support**:
+
+   Single Root I/O Virtualization (SRIOV) support is a key enhancement
+   * EMT will include SRIOV support alongside the drivers.
+   * Kubernetes device plugins or operators will be used to enable SRIOV. The SR-IOV Network Device Plugin for Kubernetes can discover and manage SR-IOV capable network devices . NVIDIA also provides an SR-IOV Network Operator  and documentation for using SR-IOV in Kubernetes.
+   * Virtual machines will leverage KubeVirt to access SR-IOV capabilities. Intel offers an "Intel Graphics SR-IOV Enablement Toolkit" with components to facilitate Graphics SR-IOV for applications using KubeVirt, allowing VMs direct access to partitioned GPU resources.
+
+1. **Documentation**:
+
+   Updated documentation will be provided, including:
+   * Instructions for deploying the NVIDIA GPU Operator.
+   * Revised guidelines for running GPU workloads on Kubernetes.
+
+1. **Monitoring and Metrics**:
+
+   GPU monitoring will be integrated into the observability stack:
+   * Tools like the NVIDIA DCGM Exporter and Intel GPU metrics will be incorporated. The NVIDIA Data Center GPU Manager (DCGM) Exporter allows for the collection of GPU metrics in Prometheus format and can be deployed in Kubernetes, often via a Helm chart . It can also be deployed as part of the NVIDIA GPU Operator . dcgm-exporter collects metrics for all GPUs on a node and can associate these metrics with specific Kubernetes pods.
+   * Dashboards will be provided for monitoring GPU usage and health.
+
+1. **Testing Across Workloads**:
+
+   The proposed dGPU support will be tested using workloads such as
+   * Geti
+   * Pallet Defect Detection(PDD)
+
+## Implementation plan
+
+The implementation is planned in two phases:
+
+### Phase 1 (for EMF 3.1 release)
+
+1. Maintain support for Intel® Arc™ iGPU and dGPU on Ubuntu 22.04
```

Comment:
iGPU or dGPU? B580 does not work. Better to list the ones already working

Reply:
Added a link to the supported GPUs. Mostly discrete.